### PR TITLE
Allow for auto allocate EIP on nat creation

### PR DESCRIPTION
**Pull request by IlyaSukhanov**

### CODE REVIEW

File: disco_aws_automation/disco_vpc_gateways.py [hunk 1]
```diff
@@ -273,3 +277,15 @@ def destroy_nat_gateways(self):
...
         wait_for_state_boto3(self.boto3_ec2.describe_nat_gateways, filter_params,
                              'NatGateways', 'deleted', 'State')
+
+        #Release EIPs of auto-
+        eips = {
+            gw['SubnetId']: gw['NatGatewayAddresses'][0]['PublicIp']
+            for gw in gateways['NatGateways']
+        }
+        for network in self.disco_vpc.networks.values():
+            nat_gateways = self.disco_vpc.get_config("{0}_nat_gateways".format(network.name))
```
Comment (IlyaSukhanov) [inline disco_aws_automation/disco_vpc_gateways.py]: OOps who ever is deleting VPC might not have access to config it was created with.

File: disco_aws_automation/disco_vpc_gateways.py [hunk 2]
```diff
@@ -226,7 +226,11 @@ def _update_nat_gateways(self, network, dry_run=False):
             if not dry_run:
                 network.delete_nat_gateways()
         else:
-            eips = [eip.strip() for eip in eips.split(",")]
+            if eips.lower() == "auto":
+                eips = [self.eip.allocate() for _ in range(len(self.network.disco_subnets.values()))]
```
Comment (IlyaSukhanov) [inline disco_aws_automation/disco_vpc_gateways.py]: Tag subnet with nat:dynonat so we know to release EIP

File: README.md [hunk 3]
```diff
@@ -1165,6 +1165,33 @@ customer has two VPNs:
...
+NAT is used to allow instances to talk to internet without having public IP's.
+The NAT must be placed into metanetwork with IGW so it can connect to internet
+itself. Then routes can be added to other metanetworks to allow them to communicate
+to internet through NAT. AWS requires assignment of EIPs to a NAT, asiaq supports
+two modes of EIP assignment: Long-lived static EIPs which are reserved using
+disco_eip.py, or temporary EIPs can be used which get released with deletion of the
+VPC. Static ips are preferred if you want to whitelist your outgoing IPs in other
+systems.
+
+To set up dynamic outbound IPs and allow all subnets to make use of NAT convigure
```
Comment (jichen-amplify) [inline README.md]: s/convigure/configured?

Comment (IlyaSukhanov) [inline README.md]: danke

**Comment by jichen-amplify** [comment]
👍

**Pull request merged**